src/details/ArborX_DetailsSortUtils.hpp

```diff
@@ -83,12 +83,11 @@ sortObjects(ExecutionSpace const &space, ViewType &view)
   using ValueType = typename ViewType::value_type;
   using CompType = Kokkos::BinOp1D<ViewType>;
 
-  Kokkos::MinMaxScalar<ValueType> result;
-  Kokkos::MinMax<ValueType> reducer(result);
-  parallel_reduce("ArborX::Sorting::find_min_max_view",
-                  Kokkos::RangePolicy<ExecutionSpace>(space, 0, n),
-                  Kokkos::Impl::min_max_functor<ViewType>(view), reducer);
-  if (result.min_val == result.max_val)
+  ValueType min_val;
+  ValueType max_val;
+  if (n > 0) // non-empty view is precondition of minMax algorithm
+    std::tie(min_val, max_val) = ArborX::minMax(space, view);
+  if ((n == 0) || (min_val >= max_val))
```

Comment:
This seems a bit convoluted. How about this:
```suggestion
  if (n == 0)
    return Kokkos::View<SizeType *, typename ViewType::device_type> permute(
                           "ArborX::Sorting::permute", 0);

  ValueType min_val;
  ValueType max_val;
  std::tie(min_val, max_val) = ArborX::minMax(space, view);
  if (min_val == max_val)
```
I would would just use `min_val == max_val` check, as `min_val >= max_val` just invites question as to when that could happen.